src/app/src/components/UserApiInfo.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { arrayOf, func, string, bool } from 'prop-types';
+
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import IconButton from '@material-ui/core/IconButton';
+import InfoIcon from '@material-ui/icons/Info';
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import {
+    userApiInfoTooltipTitles,
+    IS_NOT_SET,
+    FORBIDDEN,
+} from '../util/constants';
+
+import { fetchUserApiInfo } from '../actions/profile';
+import { userApiInfoPropType } from '../util/propTypes';
+
+const componentStyles = Object.freeze({
+    withoutMargin: Object.freeze({
+        marginBlockStart: '0',
+        marginBlockEnd: '0',
+    }),
+});
+
+const StyledTooltip = withStyles({
+    tooltip: {
+        color: 'rgba(0, 0, 0, 0.8)',
+        fontSize: '0.875rem',
+        backgroundColor: 'white',
+        border: 'solid rgba(0, 0, 0, 0.25)',
+        borderRadius: '10px',
+        padding: '10px',
+        lineHeight: '1',
+    },
+})(Tooltip);
+
+class UserAPIInfo extends Component {
+    componentDidMount() {
+        return this.props.getUserApiInfo();
+    }
+
+    render() {
+        const { error, userApiInfo, fetching } = this.props;
+
+        if (error) {
+            window.console.warn(error);
+        }
+
+        const {
+            apiCallAllowance,
+            currentCallCount,
+            renewalPeriod,
+        } = userApiInfo[0];
```

Comment:
Why do you pass an array to the front end but not dictionary? Dictionary/JSON object is more efficient.